src/mlpack/core/util/conv_to.hpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file core/util/conv_to.hpp
+ * @author Marcus Edel
+ * @author Omar Shrit
+ *
+ * A simple `conv_to` wrapper that based on the data type forwards to
+ * `coot::conv_to` or `arma::conv_to`. This file is borrowed from ensmallen.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_UTIL_CONV_TO_HPP
+#define MLPACK_CORE_UTIL_CONV_TO_HPP
+
+namespace mlpack {
+
+/**
+ * A utility class that based on the data type forwards to `coot::conv_to` or
+ * `arma::conv_to`.
+ *
+ * @tparam OutputType The data type to convert to.
+ */
+template<typename OutputType>
+class ConvTo
+{
+  public:
+
+#ifdef MLPACK_HAS_COOT
+
+    /**
+    * Convert from one matrix type to another by forwarding to `coot::conv_to`.
+    *
+    * @param input The input that is converted.
+    */
+   template<typename InputType>
+   inline static OutputType from(const InputType& input,
+                                 const typename std::enable_if_t<
+                                 coot::is_coot_type<InputType>::value ||
+                                 coot::is_coot_type<OutputType>::value>* = 0)
```

Comment:
@rcurtin now we are looking at both input and output types to cover cases such as `std::vector<T>`

Reply:
Otherwise this is ready for review :+1: